### PR TITLE
Updated TSG doc with latest traces

### PR DESCRIPTION
**Pull request by shpalani**
## Description
Updated the TSG doc as per the Traces in the latest binaries, after the fix of

https://github.com/microsoft/ebpf-for-windows/pull/2434/files
https://github.com/microsoft/ebpf-for-windows/pull/2628/files

## Testing

_Do any existing tests cover this change? N/A
Are new tests needed? No

## Documentation

_Is there any documentation impact for this change? No

### CODE REVIEW

File: docs/TroubleshootingGuide.md [hunk 1]
```diff
@@ -441,7 +448,9 @@ sc.exe stop netebpext
 sc.exe start ebpfcore
 sc.exe start netebpfext
 ```
-Then, attempt to load the program again. If this continues to fail, you will need to look further in
+Note: If `ebpfcore` fails to stop, you can attempt to restart `ebpfsvc` and then `ebpfcore`.
+
+Then, attempt to load the program again. If this continues to fail, check your ebpf program source code to see if it has incorporated valid program type and attach type. If the problem still persists, you will need to look further in
```
Comment (dthaler) [inline docs/TroubleshootingGuide.md:453]:
```suggestion
Then, attempt to load the program again. If this continues to fail, check your eBPF program source code to see if it has incorporated a valid program type and attach type. If the problem still persists, you will need to look further in
```

File: docs/TroubleshootingGuide.md [hunk 2]
```diff
@@ -411,28 +411,35 @@ There are certain errors that likely point to the eBPF client. These errors will
...
+
+[1]48D498.48D750::2023/07/18-18:49:07.123136900 [EbpfForWindowsProvider]{"Api":"DeviceIoControl","last_error":"2001(WIN=The specified driver is invalid.)","meta":{"provider":"EbpfForWindowsProvider","event":"EbpfApiError","time":"2023-07-19T01:49:07.1231369Z","cpu":1,"pid":4773016,"tid":4773712,"channel":11,"level":2,"keywords":"0x100"}}
 ```
 
-The first trace shows `ebpf_program_create` failed. Then, we see that `_ebpf_extension_client_attach_provider` fails,
-indicating that this is a NMR failure. Furthermore, we see `_ebpf_program_load_providers` which shows that the NMR
-provider load failed.
+Check the Program type's guid and Attach type's guid in the trace. Program type and Attach type must have valid guid.
+
+The first trace shows `Program type` guid is valid but `Attach type` guid is zero. Hence `ebpf_program_create` failed. The subsequent traces show 'Unable to Load Device Driver' and 'The specified driver is invalid' indicating that this is a NMR failure due to an invalid attach type.
```
Comment (dthaler) [inline docs/TroubleshootingGuide.md:441]:
```suggestion
The first trace shows the `Program type` GUID is valid but the `Attach type` GUID is zero. Hence `ebpf_program_create` failed. The subsequent traces show 'Unable to Load Device Driver' and 'The specified driver is invalid' indicating that this is an NMR failure due to an invalid attach type.
```

**Review by dthaler — DISMISSED** [review]
Ok other than editorial nits

**Review by dthaler — DISMISSED** [review]

File: docs/TroubleshootingGuide.md [hunk 3]
```diff
@@ -411,28 +411,35 @@ There are certain errors that likely point to the eBPF client. These errors will
 - `ERROR_FILE_NOT_FOUND`. This indicates that the application tried to open an eBPF program with an invalid path. This
   points to an issue within the application. The resolution is to change the path used by the application.
 
+- `Program type is not valid`. The trace shows `Program type` guid is zero. Hence `ebpf_program_create` failed. The subsequent traces show 'An invalid parameter was passed to a service or function' and 'The parameter is incorrect' indicating that the user-mode application failed to set a valid program type in the eBPF program.
```
Comment (matthewige) [inline docs/TroubleshootingGuide.md:414]: The other errors shown above are the error code values. Do we want to put ERROR_INVALID_PARAMETER here instead?

File: docs/TroubleshootingGuide.md [hunk 4]
```diff
@@ -411,28 +411,35 @@ There are certain errors that likely point to the eBPF client. These errors will
...
+
+[1]48D498.48D750::2023/07/18-18:49:07.123136900 [EbpfForWindowsProvider]{"Api":"DeviceIoControl","last_error":"2001(WIN=The specified driver is invalid.)","meta":{"provider":"EbpfForWindowsProvider","event":"EbpfApiError","time":"2023-07-19T01:49:07.1231369Z","cpu":1,"pid":4773016,"tid":4773712,"channel":11,"level":2,"keywords":"0x100"}}
 ```
 
-The first trace shows `ebpf_program_create` failed. Then, we see that `_ebpf_extension_client_attach_provider` fails,
-indicating that this is a NMR failure. Furthermore, we see `_ebpf_program_load_providers` which shows that the NMR
-provider load failed.
+Check the Program type's GUID and Attach type's GUID in the trace. Program type and Attach type must have valid GUIDs.
+
+The first trace shows the `Program type` GUID is valid but the `Attach type` GUID is zero. Hence `ebpf_program_create` failed. The subsequent traces show 'Unable to Load Device Driver' and 'The specified driver is invalid' indicating that this is an NMR failure due to an invalid attach type.
```
Comment (matthewige) [inline docs/TroubleshootingGuide.md:441]: 'Unable to Load Device Driver' and 'The specified driver is invalid'

Do you want to use the ` quote here to put these traces in the block format?

File: docs/TroubleshootingGuide.md [hunk 5]
```diff
@@ -411,28 +411,35 @@ There are certain errors that likely point to the eBPF client. These errors will
...
-"2023-05-10T20:54:44.3095699Z","cpu":1,"pid":2788,"tid":6964,"channel":11,"level":2,"keywords":"0x4"}}
+[1]48D498.48D750::2023/07/18-18:49:07.123127100 [EbpfForWindowsProvider]{"Api":"\"ebpf_core_invoke_protocol_handler\"","status":"0xC000026C(NT=Unable to Load Device Driver)","meta":{"provider":"EbpfForWindowsProvider","event":"EbpfApiError","time":"2023-07-19T01:49:07.1231271Z","cpu":1,"pid":4773016,"tid":4773712,"channel":11,"level":2,"keywords":"0x4"}}
+
+[1]48D498.48D750::2023/07/18-18:49:07.123136900 [EbpfForWindowsProvider]{"Api":"DeviceIoControl","last_error":"2001(WIN=The specified driver is invalid.)","meta":{"provider":"EbpfForWindowsProvider","event":"EbpfApiError","time":"2023-07-19T01:49:07.1231369Z","cpu":1,"pid":4773016,"tid":4773712,"channel":11,"level":2,"keywords":"0x100"}}
 ```
 
-The first trace shows `ebpf_program_create` failed. Then, we see that `_ebpf_extension_client_attach_provider` fails,
-indicating that this is a NMR failure. Furthermore, we see `_ebpf_program_load_providers` which shows that the NMR
-provider load failed.
+Check the Program type's GUID and Attach type's GUID in the trace. Program type and Attach type must have valid GUIDs.
```
Comment (matthewige) [inline docs/TroubleshootingGuide.md:439]: Do we have some list of valid program and attach type guids? If so, can we link it here? (If not, not a blocking comment, but something we should perhaps consider).

Comment (matthewige) [inline docs/TroubleshootingGuide.md:439]: Suggest: Do we want to had a bolded header for this issue (i.e NMR Attach Failure: Invalid Program or Attach Type [or something similar]).

Comment (shpalani) [reply docs/TroubleshootingGuide.md:439]: It is not documented but is present in ebpf_program_attach_type_guids.h

https://github.com/microsoft/ebpf-for-windows/blob/main/include/ebpf_program_attach_type_guids.h

File: docs/TroubleshootingGuide.md (hunk 4 above)
Comment (shpalani) [reply docs/TroubleshootingGuide.md]: Fixed. Thanks.

File: docs/TroubleshootingGuide.md (hunk 5 above)
Comment (shpalani) [reply docs/TroubleshootingGuide.md]: Added. Thanks.

Comment (shpalani) [reply docs/TroubleshootingGuide.md]: Added in a bullet.

File: docs/TroubleshootingGuide.md (hunk 3 above)
Comment (shpalani) [reply docs/TroubleshootingGuide.md]: Added. Thanks.

**Review by matthewige — DISMISSED** [review]

File: docs/TroubleshootingGuide.md [hunk 6]
```diff
@@ -425,33 +425,34 @@ There are certain errors that likely point to the eBPF client. These errors will
...
 
-[1]48D498.48D750::2023/07/18-18:49:07.123136900 [EbpfForWindowsProvider]{"Api":"DeviceIoControl","last_error":"2001(WIN=The specified driver is invalid.)","meta":{"provider":"EbpfForWindowsProvider","event":"EbpfApiError","time":"2023-07-19T01:49:07.1231369Z","cpu":1,"pid":4773016,"tid":4773712,"channel":11,"level":2,"keywords":"0x100"}}
-```
+    [1]48D498.48D750::2023/07/18-18:49:07.123136900 [EbpfForWindowsProvider]{"Api":"DeviceIoControl","last_error":"2001(WIN=The specified driver is invalid.)","meta":{"provider":"EbpfForWindowsProvider","event":"EbpfApiError","time":"2023-07-19T01:49:07.1231369Z","cpu":1,"pid":4773016,"tid":4773712,"channel":11,"level":2,"keywords":"0x100"}}
+    ```
 
-Check the Program type's GUID and Attach type's GUID in the trace. Program type and Attach type must have valid GUIDs.
+    Check the Program type's GUID and Attach type's GUID in the trace. Program type and Attach type must have valid GUIDs. Listed [here](https://github.com/microsoft/ebpf-for-windows/blob/main/include/ebpf_program_attach_type_guids.h)
```
Comment (dthaler) [inline docs/TroubleshootingGuide.md:440]: "here" violates all web design guidance. See https://www.w3.org/QA/Tips/noClickHere for example.  Also last part is a sentence fragment.

should be
```
    Check the Program type's GUID and Attach type's GUID in the trace. Program type and Attach type must have valid GUIDs as listed in [ebpf_program_attach_type_guids.h](https://github.com/microsoft/ebpf-for-windows/blob/main/include/ebpf_program_attach_type_guids.h).
```

Comment (dthaler) [reply docs/TroubleshootingGuide.md:440]: "here" violates web design guidance. See https://www.w3.org/QA/Tips/noClickHere for example.  Also last part is a sentence fragment.

```suggestion
    Check the Program type's GUID and Attach type's GUID in the trace. Program type and Attach type must have valid GUIDs as listed in [ebpf_program_attach_type_guids.h](https://github.com/microsoft/ebpf-for-windows/blob/main/include/ebpf_program_attach_type_guids.h).
```

**Review by dthaler — APPROVED** [review]

**Review by matthewige — APPROVED** [review]

**Pull request merged**